### PR TITLE
Add a section about es6 object shorthand

### PR DESCRIPTION
**Pull request by viankakrisna**
Not quite sure if the heading should be ES6 Object Shorthand or Filtering Props. Thoughts?

### CODE REVIEW

**Comment by acdlite** [comment]
Thanks for the PR! Maybe we should link to an external resource, too?

**Comment by viankakrisna** [comment]
Is this good enough? https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Operators/Object_initializer

File: docs/docs/jsx-in-depth.md [hunk 1]
```diff
@@ -245,6 +245,32 @@ function App2() {
...
+}
+```
+This is roughly equivalent to a code that looks like this:
+```js{4}
+render(){
+  const spreadProps = {
+    myPropsOne: this.props.myPropsOne,
+    myPropsTow: this.props.myPropsTwo
+  };
+  return (
```
Comment (gaearon) [inline docs/docs/jsx-in-depth.md]: I would just type props manually in the second example.

```js
<MyComponent
  myPropsOne={this.myPropsOne} 
  myPropsTwo={this.myPropsTwo} />
```

File: docs/docs/jsx-in-depth.md [hunk 2]
```diff
@@ -245,6 +245,32 @@ function App2() {
 
 Spread attributes can be useful when you are building generic containers. However, they can also make your code messy by making it easy to pass a lot of irrelevant props to components that don't care about them. We recommend that you use this syntax sparingly.
 
+### ES6 Object Shorthand
+
+If you want to filter the props that you want to pass, you can do it by using ES6 Object Shorthand in combination with object spread operator and object destructuring assignment:
```
Comment (gaearon) [inline docs/docs/jsx-in-depth.md]: Please link to MDN.

File: docs/docs/jsx-in-depth.md [hunk 3]
```diff
@@ -245,6 +245,32 @@ function App2() {
...
+
+```js{4}
+render(){
```
Comment (gaearon) [inline docs/docs/jsx-in-depth.md]: Nit: need space before `{`

File: docs/docs/jsx-in-depth.md [hunk 4]
```diff
@@ -245,6 +245,32 @@ function App2() {
...
+  const {myPropsOne, myPropsTwo} = this.props;
+  return (
+    <MyComponent {...{myPropsOne, myPropsTwo}} />
+  );
+}
+```
+This is roughly equivalent to a code that looks like this:
+```js{4}
+render(){
```
Comment (gaearon) [inline docs/docs/jsx-in-depth.md]: Nit: need space before `{`

File: docs/docs/jsx-in-depth.md [hunk 5]
```diff
@@ -245,6 +245,32 @@ function App2() {
...
 
+### ES6 Object Shorthand
+
+If you want to filter the props that you want to pass, you can do it by using ES6 Object Shorthand in combination with object spread operator and object destructuring assignment:
+
+```js{4}
+render(){
+  const {myPropsOne, myPropsTwo} = this.props;
+  return (
+    <MyComponent {...{myPropsOne, myPropsTwo}} />
```
Comment (gaearon) [inline docs/docs/jsx-in-depth.md]: Please use more realistic prop names than `myPropsOne`, `myPropsTwo`. Maybe `onClick` and `value`.

**Comment by bvaughn** [comment]
Thank you for filing this PR!

I'm sorry to be the bearer of bad news, but the documentation and source code for [reactjs.org](https://reactjs.org/) now lives in a different repository: [reactjs/reactjs.org](github.com/reactjs/reactjs.org). (For more info on why we made this move, see issue #11075.)

Would you be willing to re-open this PR on the new repo? I promise we'll review it quickly!

**Pull request closed without merging**